packages/common/src/authorization/organizationMemberAbility.ts

```diff
@@ -0,0 +1,47 @@
+import { Ability, AbilityBuilder, ForcedSubject } from '@casl/ability';
+import {
+    OrganizationMemberProfile,
+    OrganizationMemberRole,
+} from '../types/organizationMemberProfile';
+
+const actions = ['manage', 'create', 'update', 'view'] as const;
+const subjects = ['all'] as const;
+
+type PossibleAbilities = [
+    typeof actions[number],
+    (
+        | typeof subjects[number]
+        | ForcedSubject<Exclude<typeof subjects[number], 'all'>>
+    ),
+];
+
+export type OrganizationMemberAbility = Ability<PossibleAbilities>;
+
+const organizationMemberAbilities: Record<
+    OrganizationMemberRole,
+    (
+        member: OrganizationMemberProfile,
+        builder: AbilityBuilder<OrganizationMemberAbility>,
+    ) => void
+> = {
+    viewer(member, { can }) {
+        can('view', 'all');
+    },
+    editor(member, { can }) {
+        can('create', 'all');
+        can('update', 'all');
+    },
+    admin(member, { can }) {
+        can('manage', 'all');
+    },
+};
+
+export const defineAbilityForOrganizationMember = (
+    member: OrganizationMemberProfile | undefined,
```

Comment:
Should this be `user: LightdashUser | undefined` instead ? Since in the BE requests we have access to the SessionUser ( that extends the LightdashUser ) and in the FE ( in the app provider ) we have access to the user ( LightdashUser )

Reply:
Yes, that was the idea I was proposing. The LightdashUser type should have `role` and we can use that to define the ability. In the FE the only time we use  `OrganizationMemberProfile` is when the admin is listing the org members.

Reply:
Ok great, I'll move `role` from `SessionUser` to `LightdashUser`

Reply:
This is done, I've also attached the `ability` object to the `SessionUser`